sdlib/d/gc/slab.d

```diff
@@ -0,0 +1,71 @@
+module d.gc.slab;
+
+import d.gc.emap;
+import d.gc.extent;
+import d.gc.spec;
+
+enum InvalidBinID = 0xff;
+
+struct SlabAllocGeometry {
+	Extent* e;
+	void* address;
+	size_t size;
+	uint index;
+	uint sizeClass;
+
+	this(void* ptr, PageDescriptor pd) {
+		assert(pd.extent !is null, "Extent is null!");
+		assert(pd.isSlab(), "Expected a slab!");
+		assert(pd.extent.contains(ptr), "ptr not in slab!");
+
+		e = pd.extent;
```

Comment:
None of the computations in there require the extent (a very conscious decision). I'm not sure why we are extracting it. More generally, pd infos are not really something we need to provide to the caller: it already have them.